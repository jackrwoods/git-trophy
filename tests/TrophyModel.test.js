/**
 * @Author: Brogan Miner <Brogan>
 * @Date:   2019-02-25T12:04:05-08:00
 * @Email:  brogan.miner@oregonstate.edu
 * @Last modified by:   Brogan
 * @Last modified time: 2019-02-25T14:44:51-08:00
 */

// Copy everything below and put it into your test file
import React from 'react' // Import react
// eslint-disable-next-line
import { shallow, mount, render, configure } from 'enzyme'// need to import the renderer
import Adapter from 'enzyme-adapter-react-15'
configure({ adapter: new Adapter() })
// quit copying here

// eslint-disable-next-line
import TrophyModel from '../src/components/TrophyModel.js' // import the componenet to be tested

test('Can get the base', () => {
  const fakeData = new Array(52).fill(2) // make some fake data to test against
  // render the component, do this is every test (no bleeding into other tests)
  const component = shallow(
    <TrophyModel entity='Broha22' year='2018' data={fakeData} />).instance()

  // I wouldnt recommend snapshots they are too specific
  // expect(component.getBase()).toMatchSnapshot() // test against the snap folder located in __snapshots__

  // We really just seeing if it can make the object not even looking to see details about, this should probs test more.
  expect(component.getBase())
})
