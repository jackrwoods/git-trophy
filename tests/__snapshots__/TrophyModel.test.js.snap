// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can get the base 1`] = `
<mesh position={Vector3 {
      "x": 0.3571428571428571,
      "y": 0,
      "z": -0.42857142857142855,
    }
}>
  <boxGeometry
  depth={1}
  height={0.5}
  width={1.1428571428571428} />
  <meshStandardMaterial
  color={14606046}
  roughness={0.8}
  shading={1}
</mesh>
`;
